cdap-docs/examples-manual/source/examples/stream-conversion.rst

```diff
@@ -0,0 +1,163 @@
+partitions of a time-partitioned file set. These partitions can be queried with SQL. These are the
+components of the application:
+
+  - The ``events`` stream receives simple events, where each event body is a number.
+  - The ``converted`` dataset is a time-partitioned file set in Avro format.
+  - The ``StreamConversionMapReduce`` reads the last 5 minutes of events from the stream and writes them to a
+    new partition in the ``converted`` dataset.
+  - The ``StreamConversionWorkflow`` is scheduled every 5 minutes and only runs the
+    ``StreamConversionMapReduce``.
+
+Let's look at some of these components, and then run the Application and see the results.
+
+The Word Count Application
+--------------------------
+
+As in the other :ref:`examples,<examples-index>` the components
+of the Application are tied together by the class ``StreamConversionApp``:
+
+.. literalinclude:: /../../../cdap-examples/StreamConversion/src/main/java/co/cask/cdap/examples/streamconversion/StreamConversionApp.java
+     :language: java
+     :lines: 34-
+
+The interesting part is the creation of the dataset ``converted``:
+
+- It is a ``TimePartitionedFileSet``. This is an experimental new dataset type, introduced in CDAP 2.7.0. This
+  dataset manages the files in a ``FileSet`` by associating each file with a time stamp.
+- The properties are divided in two sections:
+
+  - The first set of properties configures the underlying FileSet, as documented in the
+    :ref:`FileSet,<datasets-fileset>` section.
+  - The second set of properties configures how the dataset is queryable with SQL. Here we can enable the
+    dataset for querying, and if so, we must specify Hive-specific properties for the Avro format: A SerDe,
+    and input and an output format, and an additional table property, namely the schema, for the Avro SerDe.
+
+The MapReduce Program
+---------------------
+
+In its ``beforeSubmit`` method, it determines its logical start time, and configures the ``events`` stream as its
+input and the ``converted`` dataset as its output:
+
+- This is a map-only MapReduce program, in other words, it has no reducers and the mappers directly write
+  the output, which is in Avro format::
+
+    Job job = context.getHadoopJob();
+    job.setMapperClass(StreamConversionMapper.class);
+    job.setNumReduceTasks(0);
+    job.setMapOutputKeyClass(AvroKey.class);
+    job.setMapOutputValueClass(NullWritable.class);
+    AvroJob.setOutputKeySchema(job, SCHEMA);
+
+- Based on the logical start time, determine the range of events to read from the stream::
+
+    // read 5 minutes of events from the stream, ending at the logical start time of this run
+    long logicalTime = context.getLogicalStartTime();
+    StreamBatchReadable.useStreamInput(context, "events", logicalTime - TimeUnit.MINUTES.toMillis(5), logicalTime);
+
+- Each run writes its output to a partition with the logical start time::
```

Comment:
Each MapReduce run writes its output to a partition with the logical start time::